eventio/event_filter.go

```diff
@@ -10,27 +11,118 @@ type EventFilter struct {
 	OrgGUIDs   []string
 }
 
-func (filter *EventFilter) Validate() error {
-	if filter.RangeStart == "" {
-		return fmt.Errorf(`a range start filter value is required`)
+func (filter *EventFilter) SplitByMonth() ([]EventFilter, error) {
+	dateFormat := "2006-01-02"
+
+	start, err := time.Parse(dateFormat, filter.RangeStart)
+	if err != nil {
+		return nil, err
 	}
-	if filter.RangeStop == "" {
-		return fmt.Errorf(`a range stop filter value is required`)
+	end, err := time.Parse(dateFormat, filter.RangeStop)
+	if err != nil {
+		return nil, err
 	}
-	return nil
+	var filters = []EventFilter{}
+
+	nextStartOfMonth := getNextStartOfMonth(start)
+	previousStartOfMonth := getPreviousStartOfMonth(end)
+
+	if end.Before(start) {
+		return []EventFilter{}, nil
```

Comment:
Is this not an error?

Reply:
No, if end < start => no "months" with in it.

Eg. `["9001-01-01" - "2001-01-01"]`

Reply:
I think we could refactor all of this in simpler a recursive function like
``` 
get_months(start, end) {
  if end <= start then []
  else 
     next := min(end, truncateMonth(start + "1 month"))
     [(start, next)]::get_months(next, end)
``` 

But I do not want to spend more time on this... tempted though :-D

Reply:
I'm not going to recommend recursion over iteration...better without recursion as a general rule, in my opinion :)

Reply:
OK, done

Reply:
I like it more :-D